### ISSUE TITLE
KeyError on write_fsspec of ZarrTiffStore

### ISSUE THREAD

**Issue by JoostvDoorn** [cgohlke/tifffile#249]
When converting tiff files to a zarr store we are running into a KeyError. I would think this is caused by empty tiles.

Reproduction script:
```python
import tifffile
import fsspec
import io

remote_options = {"anon": True}
urlpath = "s3://camelyon-dataset/CAMELYON17/images/patient_080_node_2.tif"

with fsspec.open(urlpath, **(remote_options or {})) as of:
    with tifffile.TiffFile(of) as tif:
        with tif.series[0].aszarr() as store:
            url, name = urlpath.rsplit("/", 1)

            of2 = io.StringIO()
            store.write_fsspec(of2, url=url)
```

Exception:
```
Traceback (most recent call last):
  File "/path/to/env/lib/python3.10/site-packages/tifffile/tifffile.py", line 13343, in _parse_key
    offset = page.dataoffsets[chunkindex]
IndexError: tuple index out of range

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/path/to/file/repro_issue_tifffile.py", line 14, in <module>
    store.write_fsspec(of2, url=url)
  File "/path/to/env/lib/python3.10/site-packages/tifffile/tifffile.py", line 13190, in write_fsspec
    keyframe, page, _, offset, bytecount = self._parse_key(key)
  File "/path/to/env/lib/python3.10/site-packages/tifffile/tifffile.py", line 13345, in _parse_key
    raise KeyError(key) from exc
KeyError: '19.0.0'
```

Version: 2024.2.12

**Comment by cgohlke** [comment]
`camelyon-dataset/CAMELYON17/images/patient_080_node_2.tif` is a Philips TIFF. In one of the TIFF pages (level 3), the number of tiles expected from the metadata does not agree with the number of stored tiles. The last row of tiles seems to be missing. It should be OK to return empty tiles instead of raising KeyError in that case.

**Comment by cgohlke** [comment]
Should be fixed in v2024.4.18
